ci: add install all npm packages after cleanup

romanrostislavovich opened the pull request:
added next gulp commands:
- `install:npm` - installing all npm packages from `package.json`. Use only with another gulp task.

Closes #90

xferra commented on script/gulp/task.test.js:
```diff
@@ -22,8 +22,8 @@ const testTrue = (test, schema, done) => {
         log.error(`FAIL: ${schema.description} ${test.description}`);
         log.error(`FILE: ${test.data.$ref}`);
         log.error(`MESSAGE: Validation failed`);
+        done();
```
It's unclear why you reverting this change after I've (probably) fixed it a few days ago. This way `done` will be never executed if the test passed, right?

romanrostislavovich replied:
was fixed

xferra commented on script/gulp/task.install.js:
```diff
@@ -0,0 +1,19 @@
+const npm = () => gulp
+    .src(config.root.package)
+    .pipe(gulpInstall({
+        args: ['--save-dev'],
```
Why save dev?

romanrostislavovich replied:
Because, all packages in ours `package.json` install as `dev`

xferra replied:
This command expected to install packages listed in `package.json`. It should not install any new packages, so this arg is not needed.

xferra commented on gulpfile.js:
```diff
@@ -33,7 +33,7 @@ gulp.task('lint', gulp.series('lint:all'));
  * @task {clean}
  */
-gulp.task('clean', gulp.series('clean:all'));
+gulp.task('clean', gulp.series('clean:all', 'install:npm'));
```
So, if `clean:build` failed somehow - `install:npm` will not be executed, right? Then it's more logical to (just schematic idea):

```javascript
task('clean', 
parallel
[
series(clean-node, install) <--- reinstall actually,
]
clean-build
clean-release
etc.
)
```

xferra replied:
BTW, as for me`install:npm` sounds like `we gonna install npm itself`, any better name?

romanrostislavovich replied:
done

xferra reviewed (changes requested):
FYI

xferra reviewed (commented):
+1

xferra commented:
@svoboda-rabstvo Please resolve conflicts.

romanrostislavovich committed:
ci: add install all npm packages after cleanup

added next gulp commands:
- `packages:install` - installing all npm packages from `package.json`. Use only with another gulp task.

added next series gulp commands:
- `packages:reinstall` - include: `clean:node` and `packages:install`

Closes #90

romanrostislavovich commented:
done

xferra reviewed (approved):
+1
